staging/src/k8s.io/kubectl/pkg/util/podutils/podutils.go

```diff
@@ -170,12 +170,10 @@ func afterOrZero(t1, t2 *metav1.Time) bool {
 }
 
 func podReadyTime(pod *corev1.Pod) *metav1.Time {
-	if IsPodReady(pod) {
-		for _, c := range pod.Status.Conditions {
-			// we only care about pod ready conditions
-			if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
-				return &c.LastTransitionTime
-			}
+	for _, c := range pod.Status.Conditions {
+		// we only care about pod ready conditions
+		if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+			return &c.LastTransitionTime
```

Comment:
or use `IsPodReady` instead, since the code above is doing the same thing as the function?

Reply:
IsPodReady returns a bool so I don't use it.

Reply:
Yeah, calling `IsPodReady` first is completely useless, I agree. It's a little sad that you can't call `getPodReadyCondition` directly, maybe make it public? Not sure how often things like this are done.

Reply:
@apelisse 
I searched the code base - this is the only place where IsPodReady is followed by a loop.

I think we can think about exposing getPodReadyCondition in the future when another instance pops up.